theano/__init__.py

```diff
@@ -126,6 +126,12 @@ def test():
         raise ImportError("The nose module is not installed."
                           " It is needed for Theano tests.")
 
+if config.device.startswith('cpu'):
+    try:
+        import theano.sandbox.mkl
+    except Exception as e:
```

Comment:
This is a very broad catch and could hide syntax error and other such things. Can you be more strict in what you catch?